### PR TITLE
Java, Go, Python, Node.js - update Guestbook & Hello World READMEs

### PR DESCRIPTION
**Pull request by kelsk**
This PR updates the Java, Go, Python, and Node samples with the same reviewed & approved content from the .NET samples (see #704 for reference).

Updates affect all **Guestbook** and **Hello World** (K8s) samples, except the Django Guestbook sample.

Total affected files: 
- 36 READMEs updated
- 24 diagram.pngs deleted
- 8 diagram.pngs updated
- 1 unused image deleted ([exclamation mark](https://github.com/GoogleCloudPlatform/cloud-code-samples/blob/d9275d5168ad81958be73d5b8c8cbc034d34955e/python/python-hello-world/img/noun_Excitement_267_3BB300.png))

A "What's in this sample" section has been added, including:
- updated Kubernetes Architecture diagram
- explanation of directory content

### CODE REVIEW

File: golang/go-guestbook/.readmes/intellij/README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 The Guestbook sample demonstrates how to deploy a Kubernetes application with a front end service and a back end service using the [Cloud Code extension for IntelliJ](https://cloud.google.com/code/docs/intellij/install?utm_source=ext&utm_medium=partner&utm_campaign=CDR_kri_gcp_cloudcodereadmes_012521&utm_content=-). When you run the application, Cloud Code uses [skaffold](https://skaffold.dev/docs/) under the hood to build an image and deploy the project's Kubernetes manifests. To learn more about Kubernetes, explore the [Kubernetes overview](https://kubernetes.io/docs/concepts/overview/). 
 
 ### Table of Contents
+* [What's in this sample](#what's-in-this-sample)
```
Comment (dinagraves) [inline golang/go-guestbook/.readmes/intellij/README.md:6]: This link doesn't seem to work

Comment (dinagraves) [reply golang/go-guestbook/.readmes/intellij/README.md:6]: Should be `#whats-in-this-sample`

File: java/java-guestbook/.readmes/intellij/README.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 The Guestbook sample demonstrates how to deploy a Kubernetes application with a front end service and a back end service using the [Cloud Code extension for IntelliJ](https://cloud.google.com/code/docs/intellij/install?utm_source=ext&utm_medium=partner&utm_campaign=CDR_kri_gcp_cloudcodereadmes_012521&utm_content=-). When you run the application, Cloud Code uses [skaffold](https://skaffold.dev/docs/) under the hood to build an image and deploy the project's Kubernetes manifests. To learn more about Kubernetes, explore the [Kubernetes overview](https://kubernetes.io/docs/concepts/overview/). 
 
 ### Table of Contents
+* [What's in this sample](#what's-in-this-sample)
```
Comment (dinagraves) [inline java/java-guestbook/.readmes/intellij/README.md:6]:
```suggestion
* [What's in this sample](#whats-in-this-sample)
```

File: nodejs/nodejs-guestbook/.readmes/intellij/README.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 The Guestbook sample demonstrates how to deploy a Kubernetes application with a front end service and a back end service using the [Cloud Code extension for IntelliJ](https://cloud.google.com/code/docs/intellij/install?utm_source=ext&utm_medium=partner&utm_campaign=CDR_kri_gcp_cloudcodereadmes_012521&utm_content=-). When you run the application, Cloud Code uses [skaffold](https://skaffold.dev/docs/) under the hood to build an image and deploy the project's Kubernetes manifests. To learn more about Kubernetes, explore the [Kubernetes overview](https://kubernetes.io/docs/concepts/overview/). 
 
 ### Table of Contents
+* [What's in this sample](#what's-in-this-sample)
```
Comment (dinagraves) [inline nodejs/nodejs-guestbook/.readmes/intellij/README.md:6]:
```suggestion
* [What's in this sample](#whats-in-this-sample)
```

File: python/python-guestbook/.readmes/intellij/README.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 The Guestbook sample demonstrates how to deploy a Kubernetes application with a front end service and a back end service using the [Cloud Code extension for IntelliJ](https://cloud.google.com/code/docs/intellij/install?utm_source=ext&utm_medium=partner&utm_campaign=CDR_kri_gcp_cloudcodereadmes_012521&utm_content=-). When you run the application, Cloud Code uses [skaffold](https://skaffold.dev/docs/) under the hood to build an image and deploy the project's Kubernetes manifests. To learn more about Kubernetes, explore the [Kubernetes overview](https://kubernetes.io/docs/concepts/overview/). 
 
 ### Table of Contents
+* [What's in this sample](#what's-in-this-sample)
```
Comment (dinagraves) [inline python/python-guestbook/.readmes/intellij/README.md:6]:
```suggestion
* [What's in this sample](#whats-in-this-sample)
```

**Review by dinagraves — CHANGES_REQUESTED** [review]

File: golang/go-guestbook/.readmes/intellij/README.md (hunk 1 above)
Comment (kelsk) [reply golang/go-guestbook/.readmes/intellij/README.md]: Fixed!

**Review by dinagraves — APPROVED** [review]

**Pull request merged**